### DISCUSSION TITLE
Yalc

### DISCUSSION THREAD

**Discussion by ElGrecode** [TanStack/query#4157] [Q&A]
I'm diving into the contributing documentation. One of the recommendations to link the package and validate a change is to use Yalc. Yalc is pretty straightforward for simple repositories, however using it a monorepo with yarn workspaces isn't simple. Yalc's documentation for advanced uses with yarn workspaces hasn't worked for this repository.

I'm trying to validate a change to `query-core`. `query-core` is a dependency of `react-query` and `react-query-persist-client`. Does anyone have a recommendation for a workflow to link `query-core` a dependency deep in the dependency graph? I'd really appreciate it and I'd be happy to update the documentation if you all think it helps more easily explain the contributing guidelines.

**Comment by TkDodo** [comment]
> I'm trying to validate a change to query-core

write a test :)

I have used yalc for getting the examples to work with local builds, and that works fine:
- do `npm run build` at the top level
- do `yalc publish in the `react-query` package
- go to the example and do `yalc add`
